packages/api/src/services/actions.ts

```diff
@@ -71,51 +72,58 @@ export function Handlers(config: Config, appState: Dependencies): Actions {
-      assert(address, "requires address");
-      assert(currency, "requires currency");
-      assert(stats[currency], "No stats for currency: " + currency);
-      return stats[currency].latest.get(address);
-    },
-    async listEmpStats(currency: CurrencySymbol = "usd") {
-      assert(currency, "requires currency");
-      assert(stats[currency], "No stats for currency: " + currency);
-      return stats[currency].latest.values();
-    },
     async tvl(addresses?: string[], currency: CurrencySymbol = "usd") {
+      addresses = lodash.castArray(addresses);
       if (addresses == null || addresses.length == 0) return queries.totalTvl(currency);
       return queries.sumTvl(addresses, currency);
     },
-    async getEmpStatsBetween(empAddress: string, start = 0, end: number = nowS(), currency: CurrencySymbol = "usd") {
+    async tvm(addresses?: string[], currency: CurrencySymbol = "usd") {
+      addresses = lodash.castArray(addresses);
+      if (addresses == null || addresses.length == 0) return queries.totalTvm(currency);
+      return queries.sumTvm(addresses, currency);
+    },
+    async tvlHistoryBetween(empAddress: string, start = 0, end: number = nowS(), currency: CurrencySymbol = "usd") {
+      assert(stats[currency], "Invalid currency type: " + currency);
+      return stats[currency].history.tvl.betweenByAddress(empAddress, start, end);
+    },
+    async tvmHistoryBetween(empAddress: string, start = 0, end: number = nowS(), currency: CurrencySymbol = "usd") {
+      assert(stats[currency], "Invalid currency type: " + currency);
+      return stats[currency].history.tvm.betweenByAddress(empAddress, start, end);
+    },
+    async tvlHistorySlice(empAddress: string, start = 0, length = 1, currency: CurrencySymbol = "usd") {
       assert(stats[currency], "Invalid currency type: " + currency);
-      assert(stats[currency].history[empAddress], "Invalid emp address: " + empAddress);
-      return stats[currency].history[empAddress].between(start, end);
+      return stats[currency].history.tvl.sliceByAddress(empAddress, start, length);
     },
-    async sliceHistoricalEmpStats(empAddress: string, start = 0, length = 1, currency: CurrencySymbol = "usd") {
+    async tvmHistorySlice(empAddress: string, start = 0, length = 1, currency: CurrencySymbol = "usd") {
       assert(stats[currency], "Invalid currency type: " + currency);
-      assert(stats[currency].history[empAddress], "Invalid emp address: " + empAddress);
       assert(length < 1000, "length must be less than 1000 samples");
-      return stats[currency].history[empAddress].slice(start, length);
+      return stats[currency].history.tvm.sliceByAddress(empAddress, start, length);
+    },
+    async listTvls(currency: CurrencySymbol = "usd") {
+      return appState.stats[currency].latest.tvl.values();
+    },
+    async listTvms(currency: CurrencySymbol = "usd") {
+      return appState.stats[currency].latest.tvm.values();
```

Comment:
there are now separate but nearly identical calls to query tvl/tvm stats both for historical and latest values